### DISCUSSION TITLE
Having an option to select which client certificate to use under the same url

### DISCUSSION THREAD

**Discussion by aroldohernandezarmas** [cypress-io/cypress#20117] [Questions and Help]
On the version [8.0.0](https://docs.cypress.io/guides/references/client-certificates) was introduced the possibility to configure cypress with client certificates for TLS client authentication. Now it is possible to specify an url pattern and a set of trusted CAs plus a set of client key/certificate pairs that can be used for the TLS handshake. I have made a search and so far couldn't find how if we specify more than one key/certificate pair per url, can we decide which one to use while running the tests.

I am trying to implement an E2E test using cypress and client cert authentication, and need more than one pair of key/certificate to verify different use cases under the same site and don't know how or if it's possible to choose specifically one pair at runtime.

Is it there any way to achieve this using only cypress?
Is it there any alternative known solution that can be used together with cypress to solve this?

**Comment by Tmodurr** [comment]
@aroldohernandezarmas have you seen any movement on this?

**Comment by acorpuz101** [comment]
I have a similar scenario and could use a way to pick which cert to use.
I've specified multiple certs in the array under `clientCertificates.[object].certs` and would like to test them both individually.
Has there been any resolution or answer to this?

**Comment by michaelsweeney5** [comment]
This is also a valid use case for tests I am writing, where I need the capability to dynamically select a client certificate in the same pipeline run, rather than running a separate pipeline with a different value in my configuration.

**Comment by manimovassagh** [comment]
I am also have the same scenario

**Comment by BreizhHardware** [comment]
1+ I have the same need
